pkg/ruler/ruler.go

```diff
@@ -881,6 +975,101 @@ func (r *Ruler) getLocalRules(userID string, rulesRequest RulesRequest) ([]*Grou
 		}
 	}
 
+	if !includeBackups {
+		return groupDescs, nil
+	}
+
+	backupGroups := r.manager.GetBackupRules(userID)
+	for _, group := range backupGroups {
```

Comment:
Can we move some code to a separate function? `getLocalRules` is like 200+ lines of code.